forage-android/src/main/java/com/joinforage/forage/android/pos/PosRefundPaymentRepository.kt

```diff
@@ -29,58 +30,65 @@ internal class PosRefundPaymentRepository(
-                        userAction = UserAction.REFUND
-                    ),
-                    posTerminalId = posTerminalId,
-                    refundParams = refundParams
+            val vaultResponse = when (
+                val response = vaultSubmitter.submit(
+                    params = PosRefundVaultSubmitterParams(
+                        baseVaultSubmitterParams = VaultSubmitterParams(
+                            encryptionKeys = encryptionKeys,
+                            idempotencyKey = paymentRef,
+                            merchantId = merchantId,
+                            path = AbstractVaultSubmitter.refundPaymentPath(paymentRef),
+                            paymentMethod = paymentMethod,
+                            userAction = UserAction.REFUND,
+                            sessionToken = sessionToken
+                        ),
+                        posTerminalId = posTerminalId,
+                        refundParams = refundParams
+                    )
                 )
-            )
-        ) {
-            is ForageApiResponse.Success -> PosRefundVaultResponse.ModelMapper.from(response.data)
-            else -> return response
-        }
+            ) {
+                is ForageApiResponse.Success -> PosRefundVaultResponse.ModelMapper.from(response.data)
+                else -> return response
+            }
 
-        val pollingResponse = pollingService.execute(
-            contentId = vaultResponse.message.contentId,
-            operationDescription = "refund of Payment $payment"
-        )
-        if (pollingResponse is ForageApiResponse.Failure) {
-            return pollingResponse
-        }
+            val pollingResponse = pollingService.execute(
+                contentId = vaultResponse.message.contentId,
+                operationDescription = "refund of Payment $payment"
+            )
+            if (pollingResponse is ForageApiResponse.Failure) {
+                return pollingResponse
+            }
 
-        val refundRef = vaultResponse.refundRef
-        return when (val refundResponse = refundService.getRefund(paymentRef, refundRef)) {
-            is ForageApiResponse.Success -> {
-                logger.i("[HTTP] Received updated Refund $refundRef for Payment $paymentRef")
-                return refundResponse
+            val refundRef = vaultResponse.refundRef
+            return when (val refundResponse = refundService.getRefund(paymentRef, refundRef)) {
+                is ForageApiResponse.Success -> {
+                    logger.i("[HTTP] Received updated Refund $refundRef for Payment $paymentRef")
+                    return refundResponse
+                }
+                else -> refundResponse
             }
-            else -> refundResponse
+        } catch (err: Exception) {
+            logger.e("Failed to refund Payment ${refundParams.paymentRef}", err)
+            return UnknownErrorApiResponse
```

Comment:
Added a catch block to ensure we handle exceptions like malformed responses from the Payments API